cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -280,6 +280,15 @@ predicate isUnreachableInCall(Node n, DataFlowCall call) { none() } // stub impl
 
 int accessPathLimit() { result = 5 }
 
+/** The unit type. */
+private newtype TUnit = TMkUnit()
+
+/** The trivial type with a single element. */
+class Unit extends TUnit {
+  /** Gets a textual representation of this element. */
+  string toString() { result = "unit" }
+}
+
```

Comment:
This should be in `DataFlowImplCommon.qll` instead.

Reply:
It can't, because for Java the `Unit` type is in the common library.

Reply:
I don't see why that is a problem, as `DataFlowImplCommon.qll` is not exposed.

Reply:
Right, that ought to work if no instance of `import java` leaks through. Currently there is such a leak (otherwise I couldn't refer to the Java version of `Unit`), so it wouldn't work without fixing that (but that should probably be separate from this PR as it could break things if someone has referenced a non-data-flow Java QL class through a `DataFlow::` prefix). Another angle on this is that we should possibly push towards all the languages adopting a common `Unit.qll`, and then use that instead of introducing such a fundamental class in `DataFlowImplCommon.qll` - it has other uses. In summary, I think the definition should still come from `DataFlowPrivate.qll`, but it should be in the form of `import semmle.code.Unit`.

Reply:
> Currently there is such a leak

:-(